src/app.ts

```diff
@@ -55,16 +55,19 @@ const listener = (): void => {
 if (process.env.NODE_ENV === 'production') {
   const server = https.createServer(
     {
-      SNICallback: (domain, cb) => {
+      SNICallback: (host, cb) => {
         // escape characters required or readFileSync will not find file
         const homePath = process.env.HOME
+        /* eslint-disable */
+        const [subDomain, domain, topLevelDomain] = host.split('.')
+        const filteredDomain = `${domain}.${topLevelDomain}`
         const secureContext = tls.createSecureContext({
           /* eslint-disable */
           key: fs.readFileSync(
-            `${homePath}/\.acme\.sh/*\.${domain}/*\.${domain}\.key`
+            `${homePath}/\.acme\.sh/*\.${filteredDomain}/*\.${filteredDomain}\.key`
           ),
           cert: fs.readFileSync(
-            `${homePath}/\.acme\.sh/*\.${domain}/*\.${domain}\.cer`
+            `${homePath}/\.acme\.sh/*\.${filteredDomain}/*\.${filteredDomain}\.cer`
```

Comment:
for websockets and other requests to work, we should be using `fullChain.cer` I believe. Could you look into the `acme` folder to find out the fullpath of the domain name?